src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,196 @@
+            PreparedStatement statement = connection.prepareStatement(query);
+            for (Role role : roles) {
+                statement.setLong(1, userId);
+                statement.setString(2, role.getRoleName().toString());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't add roles for provided userId: "
+                    + userId, e);
+        }
+    }
+
+    @Override
+    public Optional<User> getById(Long userId) {
+        String query = "SELECT * FROM users WHERE user_id = ? AND deleted = FALSE";
+        Set<Role> userRoleSet = getRoleSetByUserId(userId);
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                User searchedUser = getFromResultSet(resultSet);
+                searchedUser.setRoles(userRoleSet);
+                return Optional.of(searchedUser);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't find user by userId: " + userId, e);
+        }
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE users SET user_name = ?, login = ?, password = ? "
+                + "WHERE user_id = ? AND deleted = FALSE";
+        deleteRoles(user.getId());
+        addRoles(user.getRoles(), user.getId());
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+            return user;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update provided user: " + user, e);
+        }
+    }
+
+    private void deleteRoles(Long userId) {
+        String query = "DELETE FROM users_roles WHERE user_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, userId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete roles for provided userID: "
+                    + userId, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long userId) {
+        String query = "UPDATE users SET deleted = TRUE WHERE user_id = ? AND deleted = FALSE";
+        deleteRoles(userId);
```

Comment:
let's not delete relations with roles, it's enough that you mark user as deleted

Reply:
?